app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt

```diff
@@ -110,6 +104,8 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
     override fun onResume() {
         super.onResume()
         binding.unreadCountChip.apply { isCloseIconVisible = isChecked } // TODO: Do we need this? If yes, do we need it HERE?
+        if (onResumeIsAllowedToRefreshThreads) mainViewModel.forceRefreshThreads(filter)
+        onResumeIsAllowedToRefreshThreads = true
```

Comment:
It looks like a bit of a hack. Can't we improve this? For me the `forceRefreshThreads` should be just `updateMessages` and it's called when you launch the app and at each `onResume`.

Reply:
Has discussed, it's the best way to do it for now.